lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb

```diff
@@ -274,5 +284,9 @@ def self.embedded_method_name(fqname)
       return parts.pop, parts.pop, parts.join('/')
     end
     private_class_method :embedded_method_name
+
+    def self.loaded?(current_items, fqname)
+      current_items.any? { |item| item[:fqname].casecmp(fqname).zero? }
```

Comment:
@mkanoor this is what I was talking about... This is not necessary, imo, cause you can just allow the methods to double up

Reply:
@Fryguy I understand but we dont want to load the file multiple times and have ruby resolve it, if we can weed out the dupes before we have a final list it would be a better approach.

Reply:
@Fryguy It could also lead to circular references so we want to stop that too